packages/world/src/systems/libraries/MoveLib.sol

```diff
@@ -19,9 +18,87 @@ import { ObjectTypes } from "../../ObjectType.sol";
 
 import { Vec3, vec3 } from "../../Vec3.sol";
 import { addEnergyToLocalPool, decreasePlayerEnergy, updatePlayerEnergy } from "../../utils/EnergyUtils.sol";
-import { getMovableEntityAt, safeGetObjectTypeAt, setMovableEntityAt } from "../../utils/EntityUtils.sol";
+import {
+  getMovableEntityAt, getObjectTypeAt, safeGetObjectTypeAt, setMovableEntityAt
+} from "../../utils/EntityUtils.sol";
 
 library MoveLib {
+  function moveWithoutGravity(Vec3 playerCoord, Vec3[] memory newBaseCoords) public {
+    EntityId[] memory playerEntityIds = _removePlayerPosition(playerCoord);
+    EntityId player = playerEntityIds[0];
+
+    uint128 currentEnergy = Energy._getEnergy(player);
+
+    uint128 totalCost;
+    Vec3 currentCoord = playerCoord;
+    for (uint256 i = 0; i < newBaseCoords.length; i++) {
+      Vec3 nextBaseCoord = newBaseCoords[i];
+      _requireValidMove(currentCoord, nextBaseCoord);
+      currentCoord = nextBaseCoord;
+      totalCost += MOVE_ENERGY_COST;
+
+      if (totalCost >= currentEnergy) {
+        totalCost = currentEnergy;
+        break;
+      }
+    }
+
+    _setPlayerPosition(playerEntityIds, currentCoord);
+
+    if (totalCost > 0) {
+      decreasePlayerEnergy(player, currentCoord, totalCost);
+      addEnergyToLocalPool(currentCoord, totalCost);
+    }
```

Comment:
why dont we call `_handleAbove` in `moveWithoutGravity`, we're not applying gravity to the current player but there could be a player above them?

Reply:
nice catch!